### PR TITLE
fix #3124 . move RegistryDataConfig configuration into RegistryConfig…

### LINKED ISSUE #3124: refactor RegistryDataConfig for supporting simplify URL in registry
1. remove registryDataConfig , and move some configuration into RegistryConfig
2. remove logic : consumer url extra key。（消费者往注册中心写的URL不需要额外的key）
3. camel format support

### PR DESCRIPTION
**Pull request by cvictory**
1. remove registryDataConfig , and move some configuration into RegistryConfig
2. remove logic : consumer url extra key。（消费者往注册中心写的URL不需要额外的key）
3. camel format support

----
fix #3124

### CODE REVIEW

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java [hunk 1]
```diff
@@ -385,13 +383,11 @@ private Cluster getMergeableCluster() {
...
     private URL getRegisteredConsumerUrl(final URL consumerUrl, URL registryUrl) {
-        if (!registryUrl.getParameter(SIMPLE_CONSUMER_CONFIG_KEY, false)) {
+        if (!registryUrl.getParameter(SIMPLIFIED_KEY, false)) {
             return consumerUrl.addParameters(CATEGORY_KEY, CONSUMERS_CATEGORY,
                     CHECK_KEY, String.valueOf(false));
         } else {
-            String[] paramsToRegistry = getParamsToRegistry(DEFAULT_REGISTER_CONSUMER_KEYS,
-                    registryUrl.getParameter(EXTRA_CONSUMER_CONFIG_KEYS_KEY, new String[0]));
-            return URL.valueOf(consumerUrl, paramsToRegistry, null).addParameters(
+            return URL.valueOf(consumerUrl, DEFAULT_REGISTER_CONSUMER_KEYS, null).addParameters(
```
Comment (khanimteyaz) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java]: @cvictory would it be possible to bit explain it bit here, what we are doing here?

File: dubbo-common/src/main/java/org/apache/dubbo/common/Constants.java [hunk 2]
```diff
@@ -482,23 +482,13 @@
      * simple the registry for provider.
      * @since 2.7.0
      */
-    public static final String SIMPLE_PROVIDER_CONFIG_KEY = "simple.provider.config";
-    /**
-     * simple the registry for consumer.
-     * @since 2.7.0
-     */
-    public static final String SIMPLE_CONSUMER_CONFIG_KEY = "simple.consumer.config";
+    public static final String SIMPLIFIED_KEY = "simplified";
```
Comment (beiwei30) [inline dubbo-common/src/main/java/org/apache/dubbo/common/Constants.java]: in the future, we should categorize such constants into groups, for this case, it'd be better and more clear if we move it into RegistryConstants. @cvictory, would you mind to file one issue against this suggestion?

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java (hunk 1 above)
Comment (beiwei30) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java]: @khanimteyaz in `2.7.0` we hope to simplify URL put into the registry center.

Comment (khanimteyaz) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java]: ok. 👍

**Pull request merged**